smalltalksrc/VMMakerTests/VMObjectLayoutTests.class.st

```diff
@@ -253,6 +265,30 @@ VMObjectLayoutTests >> testObjectWithMoreThan254SlotsHasTwoHeaders [
 		self assert: numSlots equals: slots ]
 ]
 
+{ #category : #'tests - integerValues' }
+VMObjectLayoutTests >> testPositiveIntegerBorderCase1 [
+"Test of the border case when int = 2r000111111 ... . Regarding the cCode implementation, the sign bit can be lost if this test fails. The maxCInteger and numSmallIntegerTagBits guaranties the portability of the test on 32 and 64 bit computer. "
+	self deny: (memory isIntegerValue: memory maxCInteger >> memory numSmallIntegerTagBits)
+]
+
+{ #category : #'tests - integerValues' }
+VMObjectLayoutTests >> testPositiveIntegerBorderCase2 [
+"Test of the border case when int = 2r111000000 ... . Regarding the cCode implementation, the sign bit can be lost if this test fails. The maxCInteger and numSmallIntegerTagBits guaranties the portability of the test on 32 and 64 bit computer. "
+	self deny: (memory isIntegerValue: ((memory maxCInteger >> memory numSmallIntegerTagBits) bitXor: (memory maxCInteger))) "<=> isIntegerValue: (0001111) Xor (111111 ...)"
```

Comment:
What means border case, the max value possibly encoded on ...

guaranties -> guarantees

May be using a not is more readable.